File: src/main/java/atm/bloodworkxgaming/oeintegration/ExcavationEnchantment.java
package atm.bloodworkxgaming.oeintegration;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.enchantment.EnumEnchantmentType;
import net.minecraft.init.Enchantments;
import net.minecraft.inventory.EntityEquipmentSlot;
import net.minecraft.item.ItemStack;
import org.apache.commons.lang3.ArrayUtils;

/**
 * Created by jonas on 18.06.2017.
 */
public class ExcavationEnchantment extends Enchantment {


    protected ExcavationEnchantment() {
        super(Rarity.VERY_RARE, EnumEnchantmentType.DIGGER, new EntityEquipmentSlot[]{EntityEquipmentSlot.MAINHAND});
        this.setName("oreexcavation");
    }

    /**
     * Returns the minimal value of enchantability needed on the enchantment level passed.
     */
    @Override
    public int getMinEnchantability(int enchantmentLevel)
    {
        return 20;
    }

    /**
     * Returns the maximum value of enchantability nedded on the enchantment level passed.
     */
    @Override
    public int getMaxEnchantability(int enchantmentLevel)
    {
        return 50;
    }

    /**
     * Returns the maximum level that the enchantment can have.
     */
    @Override
    public int getMaxLevel()
    {
        return 1;
    }

    @Override
    public boolean canApply(ItemStack stack) {
        return ( super.canApply(stack) && !(stack.getItem().getRegistryName() != null && ArrayUtils.contains(MainConfig.enchantmentBlacklist, stack.getItem().getRegistryName().toString())))
                || (stack.getItem().getRegistryName() != null && ArrayUtils.contains(MainConfig.enchantmentWhitelist, stack.getItem().getRegistryName().toString()));
    }

    @Override
    protected boolean canApplyTogether(Enchantment ench) {
        if (MainConfig.allowMendingEnchantment){
            return super.canApplyTogether(ench);
        } else {
            return super.canApplyTogether(ench) && ench != Enchantments.MENDING;
        }
    }
}

File: src/main/kotlin/atm/bloodworkxgaming/oeintegration/enchantments/ExcavationEnchantment.kt
package atm.bloodworkxgaming.oeintegration.enchantments

import atm.bloodworkxgaming.oeintegration.MainConfig
import net.minecraft.enchantment.Enchantment
import net.minecraft.enchantment.EnchantmentType
import net.minecraft.enchantment.Enchantments
import net.minecraft.inventory.EquipmentSlotType
import net.minecraft.item.ItemStack

class ExcavationEnchantment : Enchantment(
    Rarity.VERY_RARE,
    EnchantmentType.DIGGER,
    arrayOf(EquipmentSlotType.MAINHAND)
) {

    /**
     * Returns the maximum level that the enchantment can have.
     */
    override fun getMaxLevel(): Int {
        return MainConfig.maxEnchantmentLevel.get()
    }

    override fun canApplyAtEnchantingTable(stack: ItemStack): Boolean {
        val name = stack.item.registryName.toString()
        return !MainConfig.enchantmentBlacklist.get().contains(name)
                || (super.canApplyAtEnchantingTable(stack) && MainConfig.enchantmentWhitelist.get().contains(name))
    }


    override fun checkCompatibility(other: Enchantment): Boolean {
        return if (MainConfig.allowMendingEnchantment.get()) {
            super.checkCompatibility(other)
        } else {
            super.checkCompatibility(other) && other !== Enchantments.MENDING
        }
    }
}
